onnxruntime/core/framework/error_code.cc

```diff
@@ -23,7 +24,7 @@ struct OrtStatus {
 _Check_return_ _Ret_notnull_ OrtStatus* ORT_API_CALL OrtApis::CreateStatus(OrtErrorCode code,
                                                                            _In_z_ const char* msg) NO_EXCEPTION {
   assert(!(code == 0 && msg != nullptr));
-  SafeInt<size_t> clen(nullptr == msg ? 0 : strlen(msg));
+  SafeInt<size_t> clen(nullptr == msg ? 0 : strnlen(msg, std::numeric_limits<uint16_t>::max()));
```

Comment:
This looks dangerous. Let's say msg doesn't have '\0' until count = std::numeric_limits<uint16_t>::max()-1. This will make strnlen return count. Now, we'll end up allocating sizeof(OrStatus) + count. The same is the case in other places where we use the value returned by strnlen in other functions. I suggest we cap the length to may be 512 or 1024?

Reply:
Sure - set MAX_STR_LEN as 2048.